src/linker/Linker/MarkingHelpers.cs

```diff
@@ -12,12 +11,30 @@ public MarkingHelpers (LinkContext context)
 			_context = context;
 		}
 
-		public void MarkExportedType (ExportedType type, ModuleDefinition module, in DependencyInfo reason)
+		public void MarkExportedType (ExportedType exportedType, ModuleDefinition module, in DependencyInfo reason)
 		{
-			if (!_context.Annotations.MarkProcessed (type, reason))
+			if (!_context.Annotations.MarkProcessed (exportedType, reason))
 				return;
+
+			if (reason.Kind == DependencyKind.ModuleOfExportedType) {
```

Comment:
I wouldn't condition it on a particular Kind - I think we should treat all marked exported types similarly. For example if an XML references a type forwarder I think we should keep it. And this logic in MarkStep should just go away:
https://github.com/mono/linker/blob/59eef1f943e263e876bc1194dffcdd7900906be4/src/linker/Linker.Steps/MarkStep.cs#L364